### PR TITLE
ECS container instance capacity provider

### PR DESCRIPTION
**Pull request by jamieparkinson**
https://github.com/wellcomecollection/terraform-aws-ecs-service/blob/ec2-services/modules/ec2_capacity_provider/README.md

### CODE REVIEW

File: modules/ec2_capacity_provider/asg.tf [hunk 1]
```diff
@@ -0,0 +1,103 @@
+resource "aws_autoscaling_group" "asg" {
```
Comment (kenoir) [inline modules/ec2_capacity_provider/asg.tf:1]: Used to be a problem describing ASGs in terraform, but not anymore: https://github.com/hashicorp/terraform/issues/1552

File: modules/service/service.tf [hunk 2]
```diff
@@ -6,7 +10,7 @@ resource "aws_ecs_service" "service" {
 
   // 1.4.0 is Required for EFS integration
   // Note: LATEST as of 03/07/2020 points at 1.3.0
-  platform_version = "1.4.0"
+  platform_version = var.launch_type == "FARGATE" ? "1.4.0" : null
```
Comment (kenoir) [inline modules/service/service.tf:13]: 👍

**Review by kenoir — APPROVED** [review]
Looks good to me. Have you tested both EC2 instance backed and Fargate instances?

**Comment by jamieparkinson** [comment]
Tested it by using these modules as a local source for a pipeline - an apply to the whole pipeline didn't touch the Fargate modules and worked for the service I wanted to be EC2, so I think we're safe

**Pull request merged**